crates/router/src/connector/nuvei/transformers.rs

```diff
@@ -841,57 +841,52 @@ impl<F, T>
-                        .unwrap_or_else(|| consts::NO_ERROR_MESSAGE.to_string()),
-                    reason: None,
-                    status_code: item.http_code,
+        let redirection_data = match item.data.payment_method {
+            storage_models::enums::PaymentMethod::Wallet => item
+                .response
+                .payment_option
+                .as_ref()
+                .and_then(|po| po.redirect_url.clone())
+                .map(|base_url| services::RedirectForm::from((base_url, services::Method::Get))),
+            _ => item
+                .response
+                .payment_option
+                .as_ref()
+                .and_then(|o| o.card.clone())
+                .and_then(|card| card.three_d)
+                .and_then(|three_ds| three_ds.acs_url.zip(three_ds.c_req))
+                .map(|(base_url, creq)| services::RedirectForm {
+                    endpoint: base_url,
+                    method: services::Method::Post,
+                    form_fields: std::collections::HashMap::from([("creq".to_string(), creq)]),
                 }),
-                _ => match item.response.transaction_status {
-                    Some(NuveiTransactionStatus::Error) => Err(types::ErrorResponse {
-                        code: item
-                            .response
-                            .gw_error_code
-                            .map(|c| c.to_string())
-                            .unwrap_or_else(|| consts::NO_ERROR_CODE.to_string()),
-                        message: item
-                            .response
-                            .gw_error_reason
-                            .unwrap_or_else(|| consts::NO_ERROR_MESSAGE.to_string()),
-                        reason: None,
-                        status_code: item.http_code,
-                    }),
+        };
+
+        let response = item.response;
+        Ok(Self {
+            status: get_payment_status(&response),
+            response: match response.status {
+                NuveiPaymentStatus::Error => {
+                    get_error_response(response.err_code, response.reason, item.http_code)
+                }
+                _ => match response.transaction_status {
+                    Some(NuveiTransactionStatus::Error) => get_error_response(
+                        response.gw_error_code,
+                        response.gw_error_reason,
+                        item.http_code,
+                    ),
                     _ => Ok(types::PaymentsResponseData::TransactionResponse {
-                        resource_id: item.response.transaction_id.map_or_else(
-                            || types::ResponseId::NoResponseId,
-                            types::ResponseId::ConnectorTransactionId,
-                        ),
+                        resource_id: response
+                            .transaction_id
+                            .map_or(response.order_id, Some)
+                            .map_or_else(
```

Comment:
Is this a valid scenario?,
if we don't get `transaction_id`, we should go with `response.order_id`
if we don't get `response.order_id`, we should go with empty `ConnectorTransactionId `
if we don't get `ConnectorTransactionId `, we should go with `NoResponseId `